src/spec/integration/importArchive.spec.coffee

```diff
@@ -0,0 +1,174 @@
+Promise = require 'bluebird'
+_ = require 'underscore'
+archiver = require 'archiver'
+_.mixin require('underscore-mixins')
+{Import} = require '../../lib/main'
+Config = require '../../config'
+TestHelpers = require './testhelpers'
+Excel = require 'exceljs'
+cuid = require 'cuid'
+path = require 'path'
+tmp = require 'tmp'
+fs = Promise.promisifyAll require('fs')
+# will clean temporary files even when an uncaught exception occurs
+tmp.setGracefulCleanup()
+
+createImporter = (format) ->
+  Config.importFormat = format || "csv"
+  im = new Import Config
+  im.matchBy = 'sku'
+  im.allowRemovalOfVariants = true
+  im.suppressMissingHeaderWarning = true
+  im
+
+writeXlsx = (filePath, data) ->
+  workbook = new Excel.Workbook()
+  workbook.created = new Date()
+  worksheet = workbook.addWorksheet('Products')
+  console.log "Generating Xlsx file"
+
+  data.forEach (items, index) ->
+    if index
+      worksheet.addRow items
+    else
+      headers = []
+      for i of items
+        headers.push {
+          header: items[i]
+        }
+      worksheet.columns = headers
+
+  workbook.xlsx.writeFile(filePath)
+
+CHANNEL_KEY = 'retailerA'
```

Comment:
Best practice for constants or variables to be declared at the top of the file/function.